libs/MeshKernel/include/MeshKernel/Mesh2D.hpp

```diff
@@ -352,5 +369,74 @@ namespace meshkernel
             m_facesCircumcenters.reserve(GetNumNodes());
             m_numFacesNodes.reserve(GetNumNodes());
         }
+
+        /// @brief Determine if the edges are adjacent, if so then return the start and end points (adjacent.f90)
+        void AreEdgesAdjacent(const UInt edge1, const UInt edge2, bool& areAdjacent, UInt& startNode, UInt& endNode) const;
+
+        /// @brief Find edge on the opposite side of the element
+        /// @note Currently only works for quadrilateral elements.
+        /// Will throw exception NotImplemented for non-quadrilateral element shapes.
+        UInt FindOppositeEdge(const UInt faceId, const UInt edgeId) const;
+
+        /// @brief Get the next face adjacent to the edge on the opposite side.
+        /// @param [in] faceId The starting face
+        /// @param [in] edgeId The starting edge
+        /// @return Id of neighbour face along the edge
+        UInt NextFace(const UInt faceId, const UInt edgeId) const;
+
+        /// @brief Find all elements that do no have a neighbour across any of edges.
+        void GetElementsOnDomainBoundary(std::vector<UInt>& elementsOnDomainBoundary, std::vector<UInt>& edgesOnDomainBoundary) const;
```

Comment:
this is only for quads, either make this function generic (with the number of edges an element should have) or make it explicit on the name

Reply:
Done